### PR TITLE
fix: Unify meta-data element support in mender-artifact and C++ parse…

### PR DESCRIPTION
**Pull request by SebOpsahl**
changelog: title
ticket: MEN-6199
Signed-off-by: Sebastian Opsahl <sebastian.opsahl@northern.tech>

### CODE REVIEW

**Review by kacf — APPROVED** [review]
Fantastic! I have no further comments on this one! :tada:

Next is to run the big test pipeline. I'll let you do that!

File: src/artifact/v3/header/meta_data.cpp [hunk 1]
```diff
@@ -90,11 +57,17 @@ ExpectedMetaData Parse(io::Reader &reader) {
...
+
+	return meta_data_json;
 }
 
 } // namespace meta_data
 } // namespace header
 } // namespace v3
 } // namespace artifact
-} // namespace mender
+} // namespace mender
```
Comment (oleorhagen) [inline src/artifact/v3/header/meta_data.cpp:73]: Nit: Missing newline :)

(same in the other file)

File: src/artifact/v3/header/meta_data.cpp [hunk 2]
```diff
@@ -38,39 +38,6 @@ const string empty_json_error_message =
 namespace json = mender::common::json;
 namespace log = mender::common::log;
 
-ExpectedMetaData VerifyMetaDataJson(const json::Json &json) {
```
Comment (oleorhagen) [inline src/artifact/v3/header/meta_data.cpp:41]: Not going to miss this one :laughing:

**Review by oleorhagen — APPROVED** [review]
Real nice!

**Comment by kacf** [comment]
Good work!

@SebOpsahl: As the bot suggested, this needs a cherry-pick to 4.0.x. Usually fixes should be cherry-picked to currently maintained branches.

3.5.x is not relevant since it's Golang, not C++.

**Pull request merged**